app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt

```diff
@@ -142,18 +190,39 @@ private fun ImportFilesScreen(
                 TransferTypeButtons(selectedTransferType)
                 ImportFilesTitle(Modifier.padding(vertical = Margin.Medium), titleRes = R.string.advancedSettingsTitle)
                 TransferAdvancedSettings(
-                    states = {
-                        listOf(
-                            ValidityPeriodOption.THIRTY,
-                            DownloadLimitOption.ONE,
-                            PasswordTransferOption.NONE,
-                            EmailLanguageOption.GERMAN,
-                        )
-                    },
-                    onClick = {},
+                    states = advancedOptionsCallbacks.advancedOptionsStates,
+                    onClick = ::onAdvancedOptionClicked,
                 )
             }
 
+            ValidityPeriodBottomSheet(
+                isBottomSheetVisible = { showValidityPeriodBottomSheet },
+                onOptionClicked = { advancedOptionsCallbacks.onAdvancedOptionsValueSelected(it) },
+                closeBottomSheet = { showValidityPeriodBottomSheet = false },
+                initialPosition = advancedOptionsCallbacks.initialValues.validityPeriod.ordinal,
+            )
+
+            DownloadLimitBottomSheet(
+                isBottomSheetVisible = { showDownloadLimitBottomSheet },
+                onOptionClicked = { advancedOptionsCallbacks.onAdvancedOptionsValueSelected(it) },
+                closeBottomSheet = { showDownloadLimitBottomSheet = false },
+                initialPosition = advancedOptionsCallbacks.initialValues.downloadLimit.ordinal,
+            )
+
+            PasswordOptionBottomSheet(
+                isBottomSheetVisible = { showPasswordOptionBottomSheet },
+                onOptionClicked = { advancedOptionsCallbacks.onAdvancedOptionsValueSelected(it) },
+                closeBottomSheet = { showPasswordOptionBottomSheet = false },
+                initialPosition = advancedOptionsCallbacks.initialValues.passwordOption.ordinal,
+            )
+
+            EmailLanguageBottomSheet(
+                isBottomSheetVisible = { showEmailLanguageBottomSheet },
+                onOptionClicked = { advancedOptionsCallbacks.onAdvancedOptionsValueSelected(it) },
+                closeBottomSheet = { showEmailLanguageBottomSheet = false },
+                initialPosition = advancedOptionsCallbacks.initialValues.emailLanguage.ordinal,
+            )
```

Comment:
This could probably be split in its own component regrouping all bottom sheets, and I was thinking, if we want to use a mutable state that contains the enum of the currently displayed bottom sheet, then the visibility could be handle with a single `when` that calls the correct BottomSheet component for each case.

If there's no enum mutable state, the group of bottom sheet can still go in its own component anyway

Reply:
I'll do this in the next PR to avoid conflicts as the password changes